src/lib.rs

```diff
@@ -830,7 +825,25 @@ impl From<ffi::ibv_gid> for Gid {
 
 impl From<Gid> for ffi::ibv_gid {
     fn from(mut gid: Gid) -> Self {
-        unsafe { *gid.ffi() }
+        *gid.as_mut()
+    }
+}
+
+impl AsRef<ffi::ibv_gid> for Gid {
+    fn as_ref(&self) -> &ffi::ibv_gid {
+        unsafe { self.raw.as_ptr().cast::<ffi::ibv_gid>().as_ref().unwrap() }
```

Comment:
```suggestion
        unsafe { &*self.raw.as_ptr().cast::<ffi::ibv_gid>() }
```
I don't think we need the extra safety check that `.as_ref()` implements here since we _know_ the pointer must be non-null (as it was cast from a reference).